compiler/rustc_incremental/src/persist/dirty_clean.rs

```diff
@@ -147,7 +146,28 @@ pub fn check_dirty_clean_annotations(tcx: TyCtxt<'_>) {
 
     tcx.dep_graph.with_ignore(|| {
         let mut dirty_clean_visitor = DirtyCleanVisitor { tcx, checked_attrs: Default::default() };
-        tcx.hir().visit_all_item_likes(&mut dirty_clean_visitor);
+
+        let crate_items = tcx.hir_crate_items(());
+
+        for id in crate_items.items() {
+            let item = tcx.hir().item(id);
+            dirty_clean_visitor.check_item(item.def_id, item.span);
```

Comment:
We don't need to query the HIR to get the span. We can get it using
```rust
let span = tcx.def_span(id.def_id);
dirty_clean_visitor.check_item(id.def_id, span);
```
(or, even better, only pass the `LocalDefId` and have `check_item` call `def_span`)